### ISSUE TITLE
Add underwater visual waves

### ISSUE THREAD

**Issue by ivanevans28** [depressed-pho/natural-mystic-shaders#38]
Underwater you should distort vision. I have seen this before in RSPE shaders. I really liked it.

**Comment by depressed-pho** [comment]
But in real life things don't look distorted when you are inside a body of water. That distortion effect happens only when you are looking things in water from outside (which I really want to do but is impossible in the current rendering engine).

The distortion happens when a ray of light passes from one medium to another with different [refractive indices](https://en.wikipedia.org/wiki/Refractive_index). Suppose you are outside of water and looking at a stone on the bottom. In this case there are two media, air and water, between your eyes and the stone, and their refractive indices are very different so the distortion effect will occur. But when both you and the stone are in water, there is just one medium (which is water) so refraction can't happen.

In short, the effect RSPE does is not based on the real world optics so I'm not going to do it.
